lib/open_mic.rb

```diff
@@ -0,0 +1,45 @@
+class OpenMic
+  attr_reader :location, :date, :performers
+
+  def initialize(attrs)
+    @location = attrs[:location]
+    @date = attrs[:date]
+    @performers = []
+  end
+
+  def welcome(user)
+    @performers.push(user)
+  end
+
+  def repeated_jokes?
+    # Gather all known joke ids into array and remove nesting
+    all_joke_ids = joke_ids.values.flatten!
+
+    # If all elements are unique, return false
+    if all_joke_ids.uniq.length == all_joke_ids.length
```

Comment:
I love this combo of using `uniq` and length to verify any duplicates, but I think you are over-engineering the all_joke_ids bit of it - see my notes below!